src/xlsx/helpers/xml_helpers.ts

```diff
@@ -113,29 +113,34 @@ export function escapeXml(strings: TemplateStringsArray, ...expressions): XMLStr
 }
 
 /**
- * Removes the namespace of all the xml tags in the string.
+ * Removes the escaped namespace of all the xml tags in the string.
  *
- * Eg. : "ns:test a" => "test a"
+ * Eg. : "nsTest a" => "test a"
  */
-export function removeNamespaces(query: string): string {
-  return query.replace(/[a-z0-9]+:(?=[a-z0-9]+)/gi, "");
+export function removeTagEscapedNamespaces(query: string): string {
+  return query.replace(/.*?([A-Z])(.+)/, (match, p1: string, p2: string) => {
+    return `${p1.toLowerCase()}${p2}`;
+  });
 }
 
-/**
- * Escape the namespace's colons of all the xml tags in the string.
- *
- * Eg. : "ns:test a" => "ns\\:test a"
- */
-export function escapeNamespaces(query: string): string {
-  return query.replace(/([a-z0-9]+):(?=[a-z0-9]+)/gi, "$1\\:");
+export function addEscapedNamespace(nameSpace: string, tag: string): string {
+  return nameSpace + tag.slice(0, 1).toUpperCase() + tag.slice(1);
 }
 
 /**
- * Return true if the querySelector ignores the namespaces when searching for a tag in the DOM.
+ * Replace all tags with a namespace by namespace + tag in camelCase (e.g. <x:foo> becomes <xFoo>)
  *
- * Should return true if it's running on a browser, and false if it's running on jest (jsdom).
+ * That's useful because namespaces aren't supported by the HTML specification, so it's arbitrary whether a HTML parser
+ * implementation will support namespaces in the tags or not.
  */
-export function areNamespaceIgnoredByQuerySelector() {
-  const doc = new DOMParser().parseFromString("<t:test xmlns:t='a'/>", "text/xml");
-  return doc.querySelector("test") !== null;
+export function escapeTagNamespaces(str: string): string {
+  return str.replaceAll(/(<\/?[a-zA-Z0-9]+):([a-zA-Z0-9]+)/g, (match, p1: string, p2: string) => {
```

Comment:
Maybe we could be more robust in the escaping. `xlmns:group` & `xlmnsGroup` will be equivalent once escaped. 

The main problem is that the HTML spec only support alphanumerical characters (and is case insensitive). Maybe somthing like `NAMESPACExmlsNAMESPACEgroup` ?